lib/matplotlib/backends/backend_webagg.py

```diff
@@ -63,14 +61,9 @@ class WebAggApplication(tornado.web.Application):
 
     class FavIcon(tornado.web.RequestHandler):
         def get(self):
-            image_path = os.path.join(
-                os.path.dirname(os.path.dirname(__file__)),
-                'mpl-data', 'images')
-
             self.set_header('Content-Type', 'image/png')
-            with open(os.path.join(image_path,
-                                   'matplotlib.png'), 'rb') as fd:
-                self.write(fd.read())
+            image_path = Path(rcParams["datapath"], "images", "matplotlib.png")
```

Comment:
I'm a little confused about pathlib now.  Will this work on 3.5?  I wouldn't pick on it, but the test coverage of this PR is not good, so...

Reply:
Yes, pathlib is 3.4+ (https://docs.python.org/3/library/pathlib.html) ({read,write}_{text,bytes} is 3.5+, https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_bytes).

What changed in 3.6 is that a bunch of stdlib functions that were previously only accepting string paths (open, os.path.*, etc.) started to also accept Path objects (https://www.python.org/dev/peps/pep-0519/), thus greatly decreasing the friction when using Path objects.  But that's not something we can rely on (yet).

There are quite a few places where we use the old idiom (e.g.) `with open(...) as file: ... = file.read()` which could trivially be replaced by `Path(...).read_text()`; I didn't change them (the gain is admittedly limited in that case), but only those where we also do path manipulations (because working on Path objects for *that* is much nicer than using os.path, IMO) (or where we do this multiple times in a row).

Reply:
OK, and I take it nothing trivial can be done to increase the test coverage?

Reply:
Probably not? If that makes you really uncomfortable I can revert that change, just let me know.

Reply:
No, I trust you, just asking in case you felt guilty about it...